### PR TITLE
fix for old-style workflow id

### PR DESCRIPTION
**Pull request by wai25**
Closes https://github.com/caikit/caikit/issues/216

### CODE REVIEW

File: tests/core/modules/test_module.py [hunk 1]
```diff
@@ -173,6 +173,29 @@ def test_init_and_members():
     assert config.nested.integer == 2
     assert config.nested.float == -0.123
 
+def test_backwards_compatibility_for_block():
```
Comment (joerunde) [inline tests/core/modules/test_module.py:176]: 🌶️ !

File: caikit/core/modules/config.py [hunk 2]
```diff
@@ -63,16 +63,13 @@ def __init__(self, config_dict):
 
         # 🌶️🌶️🌶️: Backwards compatibility for old-style `blocks`, `workflows`, and `resources`
         if not self.module_id:  # pylint: disable=access-member-before-definition
-            warnings.warn(
-                "No module_id found in config. Re-save the block to use module_id",
-                DeprecationWarning,
-            )  # actual warning or just log.warning?
+            log.warning("No module_id found in config. Re-save the model to use module_id.")
```
[the remark this answers is not in the document]
Comment (joerunde) [reply caikit/core/modules/config.py:66]: Interesting, could Studio instead hide deprecation warnings? 
`warnings.filterwarnings("ignore", category=DeprecationWarning)` could do it when the interpreter spins up.

I'm probably still fine with this particular change, but I don't think we can operate in a manner where we just never issue deprecation warnings

[the remark this answers is not in the document]
Comment (wai25) [reply caikit/core/modules/config.py:66]: perhaps...however I was also getting a weird pytest error on wc build. will post link in DM.

**Review by joerunde — APPROVED** [review]
lpgtm!

**Pull request merged**